cpp_test_suite/new_tests/cxx_fwd_att.cpp

```diff
@@ -803,6 +803,28 @@ class FwdAttTestSuite: public CxxTest::TestSuite
 		lo = device2->is_locked();
 		TS_ASSERT(lo == false);
 	}
+
+	void test_reading_state_forwarded_attribute(void)
+	{
+		DeviceAttribute state_attr;
+
+		// read attributes
+		TS_ASSERT_THROWS_NOTHING(state_attr = fwd_device->read_attribute("fwd_state"));
+		DevState the_state;
+		if(state_attr >> the_state)
```

Comment:
Shouldn't you extract to a  `std::vector<DevState>`?
I think this one is called now: `bool DeviceAttribute::operator >> (DevState &datum)`
And there were no (direct) changes in above operator.

Reply:
This is what we really want to test: what happens when we read a forwarded DevState attribute.
The change was indeed in DeviceAttribute::operator >> (DevState &datum) which is used internally when the user tries to read a DevState forwarded attribute.
It is actually used internally to read the orginal DevState attribute.

Reply:
My point was that the change was done in:
```
bool DeviceAttribute::operator >> (vector<DevState>& datum)
```
and not in:
```
bool DeviceAttribute::operator >> (DevState &datum)
```

But maybe the control goes internally through the vector overload anyway.

Reply:
> But maybe the control goes internally through the vector overload anyway.

I understand why you had some doubts.
There was a change in bool DeviceAttribute::operator >> (vector<DevState>& datum) and no test was added to test that.
The control goes internally to bool DeviceAttribute::operator >> (DevVarStateArray* &datum) (which was changed in this PR too) when reading a DevState forwarded attribute.
It is called from template<typename T>void FwdAttribute::set_local_attribute(DeviceAttribute &da, T *&seq_ptr), which is called by void FwdAttr::read(DeviceImpl *dev,Attribute &attr) when reading a forwarded attribute.

bool DeviceAttribute::operator >> (vector<DevState>& datum) was also changed in this PR for consistency reasons.
A test could be added to test it.

Reply:
Thanks! It's clear now.